custom_components/artnet_led/client/__init__.py

```diff
@@ -449,10 +453,45 @@ def _append_str(packet: bytearray, text: str, length: int):
         packet.extend(map(ord, padded_text))
 
     @staticmethod
-    def _consume_str(packet: bytearray, index: int, length: int) -> (str, int):
-        str_bytes = str(packet[index:index + length - 1], "ASCII")
-        string = str_bytes.split('\0')[0]
-        return string, index + length
+    def _consume_str(packet: bytearray, index: int, length: int) -> (Optional[str], int):
+        decoded_str: Optional[str] = None
+        raw_string_from_packet = packet[index:index + length]
+
+        # assume the data is ascii
+        try:
+            # if there is a NUL character in the bytearry, it terminates earlier
+            nul_terminator = re.compile(b"([^\\x00]+)")
+
+            terminated_string = nul_terminator.match(raw_string_from_packet).group(1)  \
+                if nul_terminator.match(raw_string_from_packet) else raw_string_from_packet
+
+            # remove every other control character
+            sanitized_str = re.sub(b"[^\x00-\x7F]",b"", terminated_string)
+
+            # decode
+            decoded_str = sanitized_str.decode('ascii')
```

Comment:
UTF-8 is fully compatible with standard 7-bit ASCII, so you can skip the ASCII decoding attempt.